### PR TITLE
Clean up database systems

### LINKED ISSUE #1023: Database: review db.system list
### Area(s)

area:db

Some random observations on databases in `db.system`:
- `coldfusion` is not a database, but ["a commercial rapid web-application development computing platform"](https://en.wikipedia.org/wiki/Adobe_ColdFusion). 
- `cache` is too broad and should not be used as a product name
- `FileMaker`  is an [app-development platform](https://www.claris.com/filemaker/)
- `firstsql` seems to be dead - https://dbdb.io/db/firstsqlj
- `cloudscape` is an old name of Apache Derby - https://en.wikipedia.org/wiki/Apache_Derby

most of the constants came from https://github.com/open-telemetry/opentelemetry-specification/pull/928 and it's not clear what criteria was used to add them

### Describe the solution you'd like

Let's remove obsolete, not-quite-a-database constants and clean up some obvious naming issues like `cache`.

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #1023

## Changes

- renames `cache` to `intersystems_cache` (cache is too broad)
- removes `cloudscape` in favor of `derby` "The [Apache Derby](http://db.apache.org/derby/) project is based upon Cloudscape version 10, which IBM contributed to Apache in 2004."  (see https://en.wikipedia.org/wiki/Apache_Derby, https://db.apache.org/ddlutils/databases/derby.html)
- removed `coldfusion` - it's a [web-development](https://www.adobe.com/products/coldfusion-family.html) platform, not a database. It can work with [different databases](https://helpx.adobe.com/coldfusion/configuring-administering/data-source-management-for-coldfusion.html). 
- `firstsql` - [no longer exists](https://dbdb.io/db/firstsqlj)
- `mssqlcompact` - [discontinued](https://en.wikipedia.org/wiki/SQL_Server_Compact), last released in 2011, end-of-support in 2021

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Comment by lmolkova** [comment]
I did not find any usage of `cloudscape`, `coldfusion` or `firstsql` in OpenTelemetry repositories. 
Checked `cache` in java, python and JS instrumentations - no mentions.

`mssqlcompact` is still used by .NET [OpenTelemetry.Instrumentation.EntityFrameworkCore](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/a2683bfc137ac81997458c89ce650eab2f289b1b/src/OpenTelemetry.Instrumentation.EntityFrameworkCore), but the corresponding packages for mssqlcompact are deprecated [EntityFrameworkCore.SqlServerCompact40](https://www.nuget.org/packages/EntityFrameworkCore.SqlServerCompact40), [SqlServerCompact35](https://www.nuget.org/packages/EntityFrameworkCore.SqlServerCompact35), and have <50 downloads per day.
The `other_sql` can be used as a fallback option.

File: model/registry/db.yaml [hunk 1]
```diff
@@ -229,7 +237,8 @@ groups:
               stability: experimental
             - id: mssqlcompact
               value: 'mssqlcompact'
-              brief: 'Microsoft SQL Server Compact'
+              deprecated: "Removed."
```
Comment (gregkalapos) [inline model/registry/db.yaml:240]: Maybe mention that `other_sql` can be used? Same for some of the other values here.

File: docs/attributes-registry/db.md [hunk 2]
```diff
@@ -59,61 +59,62 @@ If a parameter has no name and instead is referenced only by index, then `<key>`
...
+| `influxdb`           | InfluxDB                                           | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
+| `informix`           | Informix                                           | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
+| `ingres`             | Ingres                                             | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
+| `instantdb`          | InstantDB                                          | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
+| `interbase`          | InterBase                                          | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
+| `intersystems_cache` | InterSystems Caché                                 | ![Experimental](https://img.shields.io/badge/-experimental-blue)                                 |
```
Comment (gregkalapos) [inline docs/attributes-registry/db.md:93]: _100% off-topic comment: wow, it's really called `Caché`_

**Review by gregkalapos — APPROVED** [review]

File: docs/attributes-registry/db.md (hunk 2 above)
Comment (lmolkova) [reply docs/attributes-registry/db.md:93]: right?! there is just one caché in the world!

File: model/registry/db.yaml [hunk 3]
```diff
@@ -109,15 +114,17 @@ groups:
               stability: experimental
             - id: cloudscape
               value: 'cloudscape'
-              brief: 'Cloudscape'
+              deprecated: "Replaced by `derby`."
+              brief: "Deprecated, use `derby` instead."
```
Comment (trask) [inline model/registry/db.yaml:118]:
```suggestion
              deprecated: "Replaced by `other_sql`."
              brief: "Deprecated, use `other_sql` instead."
```

File: model/registry/db.yaml [hunk 4]
```diff
@@ -161,7 +168,8 @@ groups:
               stability: experimental
             - id: firstsql
               value: 'firstsql'
-              brief: 'FirstSQL'
+              deprecated: "Removed."
+              brief: "Deprecated, no replacement at this time."
```
Comment (trask) [inline model/registry/db.yaml:172]: Replaced by `other_sql`? (with our internal reasoning being that we don't need to list _every_ database that ever existed)

**Review by trask — APPROVED** [review]

**Pull request merged**